### PR TITLE
Add MTE-401 [vXXX] Content Blocker Updated and Shavar Github Actions

### PR DESCRIPTION
**Pull request by isabelrios**
This PR implements the work to automatically update the shavar lists used by this project. In order to do this, there are three big parts (everything is documented in detail [here](https://docs.google.com/document/d/1kA1-dr3wAxqn4Y2-rmFoofd17YU7F1fhXI4OC7mU0F4/edit#heading=h.2gnvcu6jllxv)):
1. Github action to generate the JSON files and compare with the latest version we have stored in GCP
2. Script to compare the current shavar version with the latest one on GCP
3. Github action to check if there is a new version of the shavar JSON files, if so, it creates a PR

### CODE REVIEW

File: .github/workflows/check-and-manage-shavar-files-updates.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
+name: Run script on shavar-creation-lists
+on:
+  schedule:
+    # Weekly on Monday at 00
+    - cron: '0 0 * * 1'
```
Comment (isabelrios) [inline .github/workflows/check-and-manage-shavar-files-updates.yml:5]: The job frequency would be once a month or so.

File: .github/workflows/update-content-blocker-shavar-version.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Create PR if there is a new version of shavar files
+schedule:
+    # Weekly on Monday at 3
+    - cron: '0 3 * * 1'
```
Comment (isabelrios) [inline .github/workflows/update-content-blocker-shavar-version.yml:4]: Same here, as we want this to follow previous job

File: content_blocker_update.sh [hunk 3]
```diff
@@ -4,15 +4,26 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+# Current shavar version
+CURRENT_SHAVAR_LIST_VERSION="107"
```
Comment (isabelrios) [inline content_blocker_update.sh:8]: We can start by just adding the changes in this script as a first step. We will see how it works while we wait for the shavar questions to be answered to move forward.

File: .github/workflows/check-and-manage-shavar-files-updates.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - uses: actions/checkout@v3
+      with:
+        repository: 'irios3/shavar-list-creation'
```
Comment (isabelrios) [inline .github/workflows/check-and-manage-shavar-files-updates.yml:33]: I'm using my fork becuase I had to modify the list2safebrowsing.py script to fix and issue with and how it manages some branches' names

File: .github/workflows/check-and-manage-shavar-files-updates.yml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        ls
+        mv sample_shavar_list_creation.ini shavar_list_creation.ini
+
+        echo "Run script to get JSON files"
+        python3 lists2safebrowsing.py
+
+        echo "Lists are generated"
+        ls
+
+        # Once the files are generated, check which is the latest version of these files
```
Comment (isabelrios) [inline .github/workflows/check-and-manage-shavar-files-updates.yml:55]: After syncing the branch with latest main, I was able to get all versioned files (including 108). Previously we could not get files from 98 on and the approach we took to compare files and if there were differences create a version, was wrong. Having all the versions generated simplies this script.

File: .github/workflows/check-and-manage-shavar-files-updates.yml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+          echo "$UPDATE_SHAVAR_VERSION"
+          echo "UPDATE_SHAVAR_VERSION=$UPDATE_SHAVAR_VERSION" >> $GITHUB_ENV
+        else
+          echo "There is not any new version"
+        fi
+    - name: Get disconnect-entitylist file
+      if: ${{ env.SHOULD_UPDATE_VERSION }}
+      run: |
+        git clone https://github.com/mozilla-services/shavar-prod-lists.git
+        cd shavar-prod-lists && git checkout ${{ env.UPDATE_SHAVAR_VERSION }}.0
```
Comment (isabelrios) [inline .github/workflows/check-and-manage-shavar-files-updates.yml:100]: To get the `disconnect-entitylist.json` file, we go the the new versioned branch

File: .github/workflows/check-and-manage-shavar-files-updates.yml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        cp disconnect-entitylist.json ../shavar_version_${UPDATE_SHAVAR_VERSION}
+        ls
+    - id: 'upload-folder'
+      if: ${{ env.SHOULD_UPDATE_VERSION }}
```
Comment (isabelrios) [inline .github/workflows/check-and-manage-shavar-files-updates.yml:104]: Only if we detect that the files created have a bigger version that the one stored in GCP, we updload the a new folder `shavar-lists-#version` to the repo

File: .github/workflows/check-and-manage-shavar-files-updates.yml [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+      uses: 'google-github-actions/upload-cloud-storage@v1'
+      with:
+        path: 'shavar_version_${{ env.UPDATE_SHAVAR_VERSION }}'
+        destination: 'shavar-lists-ios-public/Public/shavar-lists-v${{ env.UPDATE_SHAVAR_VERSION }}'
```
Comment (isabelrios) [inline .github/workflows/check-and-manage-shavar-files-updates.yml:108]: ⚠️ I have to remove the `v` from the folder name

File: .github/workflows/check-and-manage-shavar-files-updates.yml [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        parent: false
+    - id: 'upload-folder-1'
+      uses: 'google-github-actions/upload-cloud-storage@v1'
+      if: ${{ env.SHOULD_UPDATE_VERSION }}
```
Comment (isabelrios) [inline .github/workflows/check-and-manage-shavar-files-updates.yml:112]: If there is not a new version, the version.txt file is not modified

File: .github/workflows/update-content-blocker-shavar-version.yml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Download version file
+      run: |
+        echo "Download version.txt file"
+        curl -X GET -o "version.txt" https://storage.googleapis.com/shavar-lists-ios-public/Public/version.txt
+        new_version=`cat version.txt`
+        echo "$new_version"
+        echo "new_version=$new_version" >> $GITHUB_ENV
```
Comment (isabelrios) [inline .github/workflows/update-content-blocker-shavar-version.yml:32]: This is needed so that the branch and PR title have the shavar show the version

**Comment by isabelrios** [comment]
@lmarceau I have made changes according to the comments received by email. Things are much simplier now than before where we were comparing files. We should get all versions, new ones when available, when running the `list2safebrowsing.py` directly and compare the version with what we have. The automation should be in place with the changes in this PR.

The (hopefully only) remaing issue is related to the license of the files generated and stored in GCP, we will contact legal and see how we can manage that.

**Review by lmarceau — APPROVED** [review]
🎉

**Comment by isabelrios** [comment]
Closing this PR, the approach and so the solution may be much different after syncing with other teams involved. Slack thread with all the info

**Pull request closed without merging**